Add libsodium check in configure script

### Describe the issue
configure script lacks a check for libsodium and libsodim-dev

### Can you reliably reproduce the issue?
#### If so, please list the steps to reproduce below:
1. download BTCGPU zip file from github
2. unzip file
3. apt-get purge libsodium-dev libsodium18
4. ./autogen.sh && ./configure && make

### Expected behaviour
configure complains about missing libsodium/libsodium-dev

### Actual behaviour
configure runs fine, `make` crashes with:

```
  CXX      crypto/libbitcoinconsensus_la-aes.lo
In file included from crypto/aes.cpp:6:0:
./crypto/common.h:16:20: fatal error: sodium.h: No such file or directory
compilation terminated.
```

### What version of are you using?
bitcon-gpu-master-2f5fe4c15bb88596e7edbbda5ad702d8c075c4b7

### Machine specs:
Ubuntu Xenial (16.04 LTS) on AWS EC2 instance type `t2.small`

--- a/configure.ac
+++ b/configure.ac
@@ -719,6 +719,21 @@ if test x$use_upnp != xno; then
   )
 fi
 
+dnl Check to find the libsodium headers/libraries
+AC_CHECK_LIB(sodium, sodium_init,[],
+[AC_MSG_ERROR([The Sodium crypto library libraries not found.])]
+)
+
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
+#include <sodium.h>
+], [
+#if SODIUM_LIBRARY_VERSION_MAJOR < 9 || SODIUM_LIBRARY_VERSION_MAJOR == 9 && SODIUM_LIBRARY_VERSION_MINOR < 5
+# error
+#endif
+])],
+[AC_MSG_RESULT([checking for version of libsodium... yes])],
+[AC_MSG_ERROR([Wrong libsodium: version >= 1.0.13 required])])
+
 BITCOIN_QT_INIT
 
 dnl sets $bitcoin_enable_qt, $bitcoin_enable_qt_test, $bitcoin_enable_qt_dbus

